### ISSUE TITLE
Improve "cacheability" understanding

### ISSUE THREAD

**Issue by na--** [ironsmile/nedomi#46]
Implement a fully featured, [RFC 7234](https://tools.ietf.org/html/rfc7234) compliant comprehension of HTTP requests and responses. This way, we can know which requests can receive cached responses and which upstream responses can be cached for future requests.

Some libs which could be helpful:
- https://github.com/pquerna/cachecontrol
- https://github.com/lox/httpcache (especially the [spec tests](https://github.com/lox/httpcache/blob/master/spec_test.go))

**Comment by na--** [comment]
We might have to consider [RFC 5861](https://tools.ietf.org/html/rfc5861) and some parts of [RFC 7231](https://tools.ietf.org/html/rfc7231) as well...

**Comment by ironsmile** [comment]
Certainly! But one at a time. My guess is that trying to support everything related to caching is pretty much lost battle. We'll have to pick a feature (header) to support and implement it. Next one after that and do it incrementally.
